documentation/PreemptiveProvableAssertions.md

```
@@ -0,0 +1,446 @@
+
+To forestall a possible misunderstanding, it's worth emphasising that my preference is for protocols to strictly define and enforce simple, secure and flexible abstractions. For convenience (or as a Schelling point), the developer team may enshrine some particular use cases, but they should always be implemented at a higher layer. Therefore, this article attempts to improve flexibility in two directions:
+
+1. the mechanism (explained below) is generalised to allow arbitrarily complex assertions.
+2. the [sample implementation](https://github.com/OpenZeppelin/minimal-rollup/pull/78) exposes the "assertion" primitive to users, developers and sequencers, allowing them to build whatever systems they choose with whatever risks they accept. The protocol's job is to provide the framework and ensure risks are contained, so they only effect participants who choose to opt in.
+
+
+## Background
+
+Every 12 seconds, Ethereum selects an L1 proposer that can aggregate new L1 transactions into a block and add it to the chain. 
+
+<p align="center"><img src="./provable_assertion_images.0.png"/></p>
```

markdownlint-cli2 (0.17.2)

Images should have alternate text (alt text)

(MD045, no-alt-text)